File: training-data/ruby/hamming_distance.rb
def hamming(a, b)
  raise ArgumentError, "strands must be of equal length" unless a.length == b.length

  a.chars.zip(b.chars).count { |x, y| x != y }
end

puts hamming("GAGCCTACTAACGGGAT", "CATCGTAATGACGGCCT")
begin
  hamming("AB", "A")
rescue ArgumentError => e
  puts "error: #{e.message}"
end

File: training-data/dart/hamming_distance.dart
int hamming(String a, String b) {
  if (a.length != b.length) {
    throw ArgumentError('strands must be of equal length');
  }
  var d = 0;
  for (var i = 0; i < a.length; i++) {
    if (a.codeUnitAt(i) != b.codeUnitAt(i)) d++;
  }
  return d;
}

void main() {
  print(hamming('GAGCCTACTAACGGGAT', 'CATCGTAATGACGGCCT'));
  try {
    hamming('AB', 'A');
  } on ArgumentError catch (e) {
    print('error: ${e.message}');
  }
}
